### PR TITLE
feat: clone from global to dev reg already built images

### PR DESCRIPTION
**Pull request by andreafalzetti**
Fixes: https://github.com/okteto/okteto/issues/3921

## Description
This PR introduces a change that makes the Okteto CLI to clone an image that is already built in the global registry, to the dev registry.

## How to test 
Follow the steps in the issue.

Depending on your registry, it may be needed to clean up the test repo in your namespace.

## Demo
Internal demo: https://www.loom.com/share/ebdaff7cd2db4055941301e5298fb041

This is how the new log looks like: `DEBU[0001] Copying image 'frontend' from global to personal registry`

Screenshot 2023-08-28 at 21 56 10

### CODE REVIEW

File: pkg/registry/registry.go [hunk 1]
```diff
@@ -146,10 +146,54 @@ func (or OktetoRegistry) HasGlobalPushAccess() (bool, error) {
...
+		return "", err
+	}
+
+	i, err := descriptor.Image()
+	if err != nil {
+		return "", err
+	}
+
+	// writing the new image to the registry
+	err = or.client.Write(newRef, i)
```
Comment (teresaromero) [inline pkg/registry/registry.go:191]: What about keeping `GetDescriptor` private, and create a Public client function `WriteImageMetadata` and get the descriptor and image inside this function?

Comment (andreafalzetti) [reply pkg/registry/registry.go:191]: I think it's a good idea but I've tried to follow this suggestion and implement it that way, what I found odd is that the `WriteImageMetadata` function would require 2 inputs, one being the `imageWithDigest` which is the global image (to find the descriptor and image object), the second being `newRef` which is the metadata of the new image. 

I will leave it as is for now, perhaps a follow-up issue considering time is important now

**Comment by andreafalzetti** [comment]
I am not 100% sure about the copy of the new debug log to inform the image is being copied (See description above). In particular, if we want the global registry to be invisible, perhaps we should not mention it? But after all debug logs are mostly for debugging so it's not too bad.

File: cmd/build/v2/build.go [hunk 2]
```diff
@@ -169,9 +170,21 @@ func (bc *OktetoBuilder) Build(ctx context.Context, options *types.BuildOptions)
...
+					// if the built image belongs to global registry we clone it to the dev registry
+					// so that in can be used in dev containers (i.e. okteto up)
+					if bc.Registry.IsGlobalRegistry(imageWithDigest) {
+						oktetoLog.Debugf("Copying image '%s' from global to personal registry", svcToBuild)
+						tag := bc.Config.GetBuildHash(buildSvcInfo)
+						newImage, err := bc.Registry.CloneGlobalImageToDev(imageWithDigest, tag)
+						if err != nil {
+							return err
+						}
+						imageWithDigest = newImage
```
[the remark this answers is not in the document]
Comment (andreafalzetti) [reply cmd/build/v2/build.go:184]: I am too tired right now to actually say if this change is needed or not. Let's review and talk about this tomorrow :) cc @jLopezbarb

**Comment by jLopezbarb** [comment]
> I am not 100% sure about the copy of the new debug log to inform the image is being copied (See description above). In particular, if we want the global registry to be invisible, perhaps we should not mention it? But after all debug logs are mostly for debugging so it's not too bad.

debug logs should only be used for that purpose and the user shouldn't see them. I think it's a very useful log for debugging purposes

File: cmd/build/v2/build.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jLopezbarb) [reply cmd/build/v2/build.go:184]: Yes, it wil be the dev image given that is what `bc.Registry.CloneGlobalImageToDev` returns. Maaybe we cana renam `newImaage` to `devImage` to make it more readable

**Review by jLopezbarb — APPROVED** [review]

**Pull request merged**